### PR TITLE
Copy-edit compile docs

### PR DESCRIPTION
**Pull request by lethosor**
started as a project to clarify the cross-compilation needs for v50 (including macOS) but still a work in progress

Fixes https://github.com/DFHack/dfhack/issues/2572

### CODE REVIEW

**Comment by myk002** [comment]
Fixes #2572?

**Comment by lethosor** [comment]
Partly. I think some of those points were addressed before this PR but I'll try to get to the rest.

File: docs/dev/compile/Dependencies.rst [hunk 1]
```diff
@@ -4,55 +4,65 @@
...
-* cmake
-* Perl
-* Python
+System dependencies
+-------------------
 
-  * Sphinx
-* git (required for `contributions <https://github.com/DFHack/dfhack/pulls>`_)
+
+* CMake
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:32]:
```suggestion
* CMake (v3.21 is recommended)
```

File: docs/dev/compile/Dependencies.rst [hunk 2]
```diff
@@ -4,55 +4,65 @@
...
+* build system (e.g. gcc & ninja, or Visual Studio)
+* Perl (for code generation)
+* Python (for documentation)
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:35]:
```suggestion
* Python 3 (for `documentation <documentation>`)
```

File: docs/dev/compile/Dependencies.rst [hunk 3]
```diff
@@ -4,55 +4,65 @@
...
-* Python
+System dependencies
+-------------------
 
-  * Sphinx
-* git (required for `contributions <https://github.com/DFHack/dfhack/pulls>`_)
+
+* CMake
+* build system (e.g. gcc & ninja, or Visual Studio)
+* Perl (for code generation)
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:34]:
```suggestion
* Perl 5 (for code generation)
```

File: docs/dev/compile/Dependencies.rst [hunk 4]
```diff
@@ -4,55 +4,65 @@
...
 * ccache (**optional**, but recommended to improve build times)
 * OpenGL headers (**optional**: to build `stonesense`)
+* SDL (libsdl 1.2, not sdl2)
 * zlib (compression library used for `xlsxreader-api` -> `quickfort`)
-* build system (e.g. gcc & ninja, or Visual Studio)
 
 ..
     maybe the below should be talked about next to the bullet point??
 
 **SDL** is used as an injection point which you can see more about in DFHack's `architectural <architectural-diagrams>` documentation & diagrams.
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:45]:
```suggestion
**SDL** is used as an injection point which you can see more about in DFHack's `architectural documentation & diagrams <architectural-diagrams>`.
```

File: docs/dev/compile/Dependencies.rst [hunk 5]
```diff
@@ -104,23 +117,27 @@ through your package manager, if possible.
...
-  * This is definitely required on Ubuntu/Debian, check if using a different distribution.
+  * If you have installed a non-default version of GCC - for example, GCC 4.8 on a
+    distribution that defaults to 5.x - you may need to add the version number to
+    the multilib packages.
+
+    * For example, ``gcc-4.8-multilib`` and ``g++-4.8-multilib`` if installing for GCC 4.8
+      on a system that uses a later GCC version.
+    * This is definitely required on Ubuntu/Debian, check if using a different distribution.
+
+  * Note that installing a 32-bit GCC on 64-bit systems (e.g. ``gcc:i386`` on
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:134]:
```suggestion
  * Note that installing 32-bit GCC on 64-bit systems (e.g. ``gcc:i386`` on
```

File: docs/dev/compile/Dependencies.rst [hunk 6]
```diff
@@ -82,13 +92,16 @@ Here are some package install commands for various distributions:
...
+distributions. Using exactly GCC 4.8 has the benefit of avoiding `libstdc++
+compatibility issues <linux-incompatible-libstdcxx>`, but can be hard to obtain
+on modern distributions, and is usually not needed because the ``dfhack``
+launcher script automatically works around libstd++ compatibility issues. As
+long as your system-provided GCC is new enough, it should work. Note that
+extremely new GCC versions may not have been used to build DFHack yet, so if you
+run into issues with these, please let us know (e.g. by opening a GitHub issue).
 
 Before you can build anything, you'll also need ``cmake``. It is advisable to
 also get ``ccmake`` on distributions that split the cmake package into multiple
```
Comment (cppcooper) [inline docs/dev/compile/Dependencies.rst:107]:
```suggestion
We recommend installing ``ccmake`` on distributions that split the cmake package into multiple
```

File: docs/dev/compile/Dependencies.rst (hunk 5 above)
Comment (lethosor) [reply docs/dev/compile/Dependencies.rst:134]: Disagree - it's similar to "an installation of GCC"

File: docs/dev/compile/Dependencies.rst (hunk 1 above)
Comment (lethosor) [reply docs/dev/compile/Dependencies.rst:32]: or newer

File: docs/dev/compile/Dependencies.rst (hunk 5 above)
Comment (cppcooper) [reply docs/dev/compile/Dependencies.rst:134]: I think it's actually more like "installing a 32-bit Windows"

Comment (lethosor) [reply docs/dev/compile/Dependencies.rst:134]: You can have more than one installation of GCC on a system. You can't have more than one installation of Windows on a system (ok, disregarding VMs, dual-booting, etc., but those aren't relevant in the context of compiling DFHack on _one_ system). I can reword this, though - maybe "installing a 32-bit version of GCC"

Comment (cppcooper) [reply docs/dev/compile/Dependencies.rst:134]: That sounds fine in my ear.

**Comment by cppcooper** [comment]
I guess you can't make a suggestion for a line that doesn't have a diff.. but I think the line `choco install cmake` should be replaced with `choco install cmake.portable`

I noticed that the former doesn't create shims (or presumably update the PATH, assuming a cmake binary actually even got installed which I did not dig for in C:\ProgramData)

**Pull request merged**